### PR TITLE
[PLUGIN-691] Fix NPE when working with nullable time field in MSSQL

### PR DESCRIPTION
**Pull request by MEseifan**
Cherrypick of #111

### CODE REVIEW

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java [hunk 1]
```diff
@@ -21,15 +21,15 @@
...
 import java.sql.SQLException;
+import java.sql.Timestamp;
 import java.sql.Types;
 import java.util.List;
 
 /**
- * SQL Server Source implementation {@link org.apache.hadoop.mapreduce.lib.db.DBWritable} and
- * {@link org.apache.hadoop.io.Writable}.
+ * SQL Server Source implementation {@link org.apache.hadoop.mapreduce.lib.db.DBWritable} and {@link
+ * org.apache.hadoop.io.Writable}.
```
Comment (yaojiefeng) [inline mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java:32]: nit - put @link in the same line

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java [hunk 2]
```diff
@@ -42,14 +42,17 @@ public SqlServerSourceDBRecord() {
   }
 
   @Override
-  protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
-                             int columnIndex, int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
+  protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder,
+      Schema.Field field,
```
Comment (yaojiefeng) [inline mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java]: indentation is off

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java [hunk 3]
```diff
@@ -42,14 +42,17 @@ public SqlServerSourceDBRecord() {
...
+      int columnIndex, int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
     switch (sqlType) {
       case Types.TIME:
         // Handle reading SQL Server 'TIME' data type to avoid accuracy loss.
         // 'TIME' data type has the accuracy of 100 nanoseconds(1 millisecond in Informatica)
         // but reading via 'getTime' and 'getObject' will round value to second.
-        recordBuilder.setTime(field.getName(), resultSet.getTimestamp(columnIndex).toLocalDateTime().toLocalTime());
+        final Timestamp timestamp = resultSet.getTimestamp(columnIndex);
```
Comment (yaojiefeng) [inline mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java]: we don't need final here

**Review by yaojiefeng — APPROVED** [review]
LGTM, Just some minor comments

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java (hunk 1 above)
Comment (MEseifan) [reply mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java:32]: done

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java (hunk 2 above)
Comment (MEseifan) [reply mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java]: reverted change

File: mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java (hunk 3 above)
Comment (MEseifan) [reply mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java]: done

**Pull request merged**